cmd/common/image/build.go

```diff
@@ -17,80 +17,202 @@ package image
+			}
+
+			fFlag := cmd.Flags().Lookup("file")
+			opts.fileChanged = fFlag.Changed
+
+			opts.path = args[0]
+
+			return runBuild(opts)
+		},
 	}
 
-	cmd.Flags().String("prog-amd64", "", "path to the amd64 variant of the eBPF program")
-	cmd.Flags().String("prog-arm64", "", "path to the arm64 variant of the eBPF program")
-	cmd.Flags().String("definition", "", "path to the definition file")
+	cmd.Flags().StringVarP(&opts.file, "file", "f", "build.yaml", "path to build.yaml")
+	cmd.Flags().BoolVarP(&opts.local, "local", "l", false, "build using local tools")
+	cmd.Flags().StringVarP(&opts.image, "tag", "t", "", "name for the built image (format name:tag)")
+	cmd.Flags().StringVar(&opts.builderImage, "builder-image", DEFAULT_BUILDER_IMAGE, "builder image to use")
 
 	return cmd
 }
 
-func runBuild(cmd *cobra.Command, args []string) error {
-	if len(args) != 1 {
-		return fmt.Errorf("expected exactly one unnamed argument")
+type imageIndexOpts struct {
+	progAmd64FilePath  string
+	progArm64FilePath  string
+	definitionFilePath string
+	image              string
+}
+
+func runBuild(opts *cmdOpts) error {
+	conf := &buildFile{
+		EBPFProgram: "program.bpf.c",
+		Definition:  "definition.yaml",
 	}
-	o := &buildOptions{
-		progAmd64FilePath:  cmd.Flag("prog-amd64").Value.String(),
-		progArm64FilePath:  cmd.Flag("prog-arm64").Value.String(),
-		definitionFilePath: cmd.Flag("definition").Value.String(),
-		image:              args[0],
+
+	var buildContent []byte
+	var err error
+
+	if opts.fileChanged {
+		buildContent, err = os.ReadFile(opts.file)
+		if err != nil {
+			return fmt.Errorf("reading build file: %w", err)
+		}
+	} else {
+		buildContent, err = os.ReadFile(filepath.Join(opts.path, opts.file))
+		if err != nil && !errors.Is(err, os.ErrNotExist) {
+			return fmt.Errorf("reading build file: %w", err)
+		}
 	}
 
-	if o.progArm64FilePath == "" && o.progAmd64FilePath == "" {
-		return fmt.Errorf("at least one of --prog-amd64 or --prog-arm64 must be specified")
+	if len(buildContent) > 0 {
+		if err := yaml.Unmarshal(buildContent, conf); err != nil {
+			return fmt.Errorf("unmarshal build.yaml: %w", err)
+		}
 	}
 
-	ociStore, err := utils.GetLocalOciStore()
+	// make a temp folder to store the build results
+	tmpDir, err := os.MkdirTemp("", "gadget-build")
 	if err != nil {
-		return fmt.Errorf("get oci store: %w", err)
+		return fmt.Errorf("create temp dir: %w", err)
 	}
+	defer os.RemoveAll(tmpDir)
 
-	indexDesc, err := createImageIndex(ociStore, o)
-	if err != nil {
-		return fmt.Errorf("create image index: %w", err)
+	if opts.path != "." {
+		cwd, err := os.Getwd()
+		if err != nil {
+			return fmt.Errorf("getting current directory: %w", err)
+		}
+		defer os.Chdir(cwd)
+
+		if err := os.Chdir(opts.path); err != nil {
+			return fmt.Errorf("changing directory: %w", err)
+		}
+	}
+
+	if opts.local {
+		buildCmd := exec.Command(
+			"/bin/sh", "-c",
+			string(buildScript),
+			"", // TODO: why is it needed?
```

Comment:
It's like the `argv[0]` of `execve`:
```
$ sh -c 'echo $0; echo $*' one two three
one
two three
```

```suggestion
			"ig-build-script", // Synthesize argv[0]
```

Reply:
Thanks for the explanation. Also to @burak-ok that explained it in another comment too. Updated it!